src/death/corpses.ts

```diff
@@ -188,16 +189,37 @@ const CORPSE_MAX_AGE_MILLIS = CORPSE_MAX_AGE_HOURS * 60 * 60 * 1000;
 registerEvent(ChunkUnloadEvent, async (event) => {
   for (const entity of event.chunk.entities) {
     if (isCorpse(entity)) {
-      const age = getCorpseAge(entity);
-      if (age === undefined) {
-        entity.remove();
-      } else if (age > CORPSE_MAX_AGE_MILLIS) {
-        entity.remove();
-      }
+      tryDespawnCorpse(entity);
     }
   }
 });
 
+// Remove old corpses (Some chunks might never unload so this is the only way to despawn)
+// We dont need to despawn those armorstands that often, so long delay is ok
+const DESPAWN_INTERVAL_HOURS = 3;
+setInterval(() => {
+  checkAllCorpsesForDespawning();
+}, DESPAWN_INTERVAL_HOURS * 60 * 60 * 1000);
+
+function checkAllCorpsesForDespawning() {
```

Comment:
This could cause lag if there are many armor stands, but might be good enough for now. In future, consider running these kind of tasks when there are unlikely to be any players.